documentation/general/design/dash-sonic-hld.md

```diff
@@ -0,0 +1,467 @@
+
+## 1.3 Warm Restart requirements
+Warm-restart support is not considered in Phase 1. TBD
+
+## 1.4 Scaling requirements
+Following are the minimal scaling requirements
+| Item                     | Expected value              |
+|--------------------------|-----------------------------|
+| VNETs                    | 1024                     |
+| ENI                      | 32 Per Card              |
+| Routes per ENI           | 100k                     |
+| NSGs per ENI             | 6                        |
+| ACLs per ENI             | 6x100K prefixes          |
+| ACLs per ENI             | 6x10K SRC/DST ports      |
+| CA-PA Mappings           | 10M                      |
+| Active Connections/ENI   | 1M                       |
+
+# 2 Packet Flows
+	
+The following section captures at a high-level on the VNET packet flow. Detailed lookup and pipeline behavior can be referenced *here*.
+
+## 2.1 Outbound packet processing pipeline
+	
+  ![dash-outbound](https://github.com/prsunny/DASH/blob/main/Assets/pkt_flow_outbound.png)
+	
+Based on the incoming packet's VNI matched against the reserved VNI assigned for VM->Appliance, the pipeline shall set the direction as TX(Outbound) and using the inner src-mac, maps to the corresponding ENI.The incoming packet will always be vxlan encapped and outer dst-ip is the appliance VIP. The pipeline shall parse the VNI, and for VM traffic, the VNI shall be a special reserved VNI. Everything else shall be treated as as network traffic(RX). Pipeline shall use VNI to differentiate the traffic to be VM (Inbound) or Network (Outbound).
+
+In the outbound flow, the appliance shall assume it is the first appliance to apply policy. It applies the outbound ACLs in three stages (VNIC, Subnet and VNET), processed in order, with the outcome being the most restrictive of the three ACLs combined. 
+
+After the ACL stage, it does LPM routing based on the inner dst-ip and applies the respective action (encap, subsequent CA-PA mapping). Finally, update the connection tracking table for both inbound and outbound. 
+	
+## 2.2 Inbound packet processing pipeline
+	
+   ![dash-intbound](https://github.com/prsunny/DASH/blob/main/Assets/pkt_flw_inbound.png)
+
+Based on the incoming packet's VNI, if it does not match against any reserved VNI, the pipeline shall set the direction as RX(Inbound) and using the inner dst-mac, maps to the corresponding ENI. In the inbound flow, Routing (LPM) lookup happens first based on the inner dst-ip and does a CA-PA validation based on the mapping. After LPM is the three stage ACL, processed in order. ACLs can have multiple src/dst IP ranges or port ranges as match criteria.
+	
+It is worth noting that CA-PA mapping table shall be used for both encap and decap process
+	
+# 3 Modules Design
+
+The following are the schema changes. The NorthBound APIs shall be defined as sonic-yang in compliance to [yang-guideline](https://github.com/Azure/SONiC/blob/master/doc/mgmt/SONiC_YANG_Model_Guidelines.md)
```

Comment:
Will this document eventually describe the yang schema, or at least provide a link to yang source files?

Reply:
Yes, the document is in its initial form. Will have more updates coming.